### PR TITLE
Indicate that ILM is on by default and add new settings

### LINKED ISSUE #11347: ILM documenation update
Index Lifecycle Management (ILM) documentation needs to be updated.

For example:
https://www.elastic.co/guide/en/beats/filebeat/7.x/ilm.html
https://www.elastic.co/guide/en/beats/metricbeat/7.x/ilm.html

In 7.x ILM is automatic, no setup required. The documentation should reflect this and describe the default policy. 

In 6.7 and 7.x ILM is no longer consider BETA (at least from Elasticsearch's perspective).

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #11347 

Adds new ILM settings plus reworks the conceptual content. Also fixed a minor nit for people who are bothered by the mixture of "configure" and "set up" in the TOC. 

Marking "in progress" because I haven't tested the settings yet. You can feel free to comment whenever you're able.

TODO:
- [ ] Decide who should update the reference.yml.

### CODE REVIEW

**Comment by dedemorton** [comment]
@bmorelli25 FYI that I'm updating the ILM docs. I read somewhere that ILM might be off by default for APM. If that's true, and you reuse this content, things are gonna get interesting....

**Comment by bmorelli25** [comment]
Thanks for the heads up. ILM is not in 7.0 for APM Server. Things will get interesting later though...

File: libbeat/docs/shared-ilm.asciidoc [hunk 1]
```diff
@@ -1,138 +1,114 @@
...
+{beatname_uc} loads the default policy automatically and applies it to any
+indices created by {beatname_uc}.
+
+You can view and edit the policy in the *Index lifecycle policies* UI in {kib}.
+For more information about working with the UI, see
+{kibana-ref}/index-lifecycle-policies.html[Index lifecyle policies].
+
+// REVIEWERS: I'm not going to describe the default policy as requested in the
+// GitHub issue because the settings might change. If users want to know what
+// the policy settings are, they should view the policy in Kibana. Sound fair?
```
Comment (urso) [inline libbeat/docs/shared-ilm.asciidoc]: +1

File: libbeat/docs/shared-ilm.asciidoc [hunk 2]
```diff
@@ -1,138 +1,114 @@
...
 ----
 <1> Date math is supported here. For more information, see
 {ref}/indices-rollover-index.html#_using_date_math_with_the_rollover_api[Using
 date math with the rollover API].
 
-NOTE: If you modify the `rollover_alias` or `pattern` settings after loading the
-index template, you must overwrite the template to apply the changes.
---
+// REVIEWERS: Rather than showing the default for setup.ilm.pattern, let's show
+// something different than the default. What would you suggest here?
```
Comment (urso) [inline libbeat/docs/shared-ilm.asciidoc]: Hm... I think I'd stick with the default or do not show it in the example. It's the recommended setting and I'd rather not want users to change it.
Maybe remove the 'date' part from the pattern if you want it to differ from the default.

File: libbeat/docs/shared-ilm.asciidoc [hunk 3]
```diff
@@ -1,138 +1,114 @@
...
 
-. Load the default policy into {es}. You can either use the `setup` command to
-load the policy without modifying it, or modify the policy and load it manually.
-+
---
-To use the setup command, run:
+// TODO: The reference.yml is missing some of these options and incorrectly
+// refers to the "elasticsearch.output.index" (it should be output.elasticsearch.index)
+//
+// Should I also update the reference.yml, or is it too late for 7.0?
```
Comment (urso) [inline libbeat/docs/shared-ilm.asciidoc]: Please update. There still will be other builds produced from the 7.0 branch in the future.

File: libbeat/docs/shared-ilm.asciidoc [hunk 4]
```diff
@@ -1,138 +1,114 @@
...
+[float]
+[[setup-ilm-option]]
+==== `setup.ilm.enabled`
 
---
+Enables or disables index lifecycle management on any new indices created by
+{beatname_uc}. Valid values are `true`, `false`, and `auto`. When `auto` (the
+default) is specified, {beatname_uc} automatically uses index lifecycle
+management if the feature is enabled in {es} and has the required license;
+otherwise, {beatname_uc} creates daily indices.
```
Comment (urso) [inline libbeat/docs/shared-ilm.asciidoc]: 'auto' only enables ILM if ES >= 7.0.0.

File: libbeat/docs/shared-ilm.asciidoc [hunk 5]
```diff
@@ -1,138 +1,114 @@
...
-set `ilm.enabled`, or the settings specified in the configuration file will be
-overwritten.
+[float]
+[[setup-ilm-policy_file-option]]
+==== `setup.ilm.policy_file`
 
-This configuration results in a managed index named something like
-+customname-{localdate}-000001+ and the following index settings:
+The path to a JSON file that contains a lifecycle policy configuration. Use this
+setting to load your own lifecycle policy.
```
Comment (urso) [inline libbeat/docs/shared-ilm.asciidoc]: How about adding a link to the ES ILM docs?

File: libbeat/docs/shared-ilm.asciidoc [hunk 6]
```diff
@@ -1,138 +1,114 @@
...
+
+[float]
+[[setup-ilm-overwrite-option]]
+==== `setup.ilm.overwrite`
 
+When set to `true`, the lifecycle policy is overwritten at startup. The default
+is `false`.
 
+// REVIEWERS: I think the section about "advanced" settings is no longer required
+// now that we use the alias to override the index setting, so I am deleting it.
```
Comment (urso) [inline libbeat/docs/shared-ilm.asciidoc]: 👍

File: libbeat/docs/shared-ilm.asciidoc (hunk 2 above)
Comment (dedemorton) [reply libbeat/docs/shared-ilm.asciidoc]: Leaving as-is (to show the default).

File: libbeat/docs/shared-ilm.asciidoc (hunk 3 above)
Comment (dedemorton) [reply libbeat/docs/shared-ilm.asciidoc]: Created https://github.com/elastic/beats/issues/11575 to track addition of ILM options

**Pull request merged**